staging/src/k8s.io/apiserver/pkg/apis/config/validation/validation.go

```diff
@@ -233,14 +238,22 @@ func validateKMSAPIVersion(c *config.KMSConfiguration, fieldPath *field.Path) fi
 	return allErrs
 }
 
-func validateKMSConfigName(c *config.KMSConfiguration, fieldPath *field.Path) field.ErrorList {
+func validateKMSConfigName(c *config.KMSConfiguration, fieldPath *field.Path, kmsProviderNames sets.String) field.ErrorList {
 	allErrs := field.ErrorList{}
 	if c.Name == "" {
 		allErrs = append(allErrs, field.Required(fieldPath, fmt.Sprintf(mandatoryFieldErrFmt, "name", "provider")))
 	}
 
-	if c.APIVersion != "v1" && strings.Contains(c.Name, ":") {
-		allErrs = append(allErrs, field.Invalid(fieldPath, c.Name, fmt.Sprintf(invalidKMSConfigNameErrFmt, c.Name)))
+	if c.APIVersion != "v1" {
+		// kms v2 providers are not allowed to have a ":" in their name
+		if strings.Contains(c.Name, ":") {
+			allErrs = append(allErrs, field.Invalid(fieldPath, c.Name, fmt.Sprintf(invalidKMSConfigNameErrFmt, c.Name)))
+		}
+		// kms v2 providers name should be unique across all kms providers (v1 and v2)
+		// TODO(aramase): make this check for v1 and v2 (all kms versions) once https://github.com/kubernetes/kubernetes/pull/113529 is merged
```

Comment:
I see https://github.com/kubernetes/kubernetes/pull/113529 has been merged. I can make this change in a follow-up PR. Let me know!

Reply:
Opened https://github.com/kubernetes/kubernetes/pull/113697 as a follow-up. Changes are in https://github.com/kubernetes/kubernetes/pull/113697/commits/47f8c4bec63a2c4d6406cd615b41cd16f12be434